horovod/run/common/util/settings.py

```diff
@@ -48,4 +47,9 @@ def __init__(self, verbose=0, ssh_port=None, key=None, timeout=None,
         self.num_proc = num_proc
         self.hosts = hosts
         self.output_filename = output_filename
+        self.command = None
+        self.run_func_mode = None
+
+    def set_command(self, command, run_func_mode):
```

Comment:
I think it'd be cleaner to just decouple the command from the settings and pass it in as a separate parameter.  The `run_func_mode` property can be set in the constructor because you know its value from `args.run_func`.